### ISSUE TITLE
[Improvement] Use tensorflow global_step instead of counter

### ISSUE THREAD

**Issue by howardyclo** [yenchenlin/pix2pix-tensorflow#15]
Hi, Yen-Chen, first of all, this is really a great work :-)
I've used it and have an advice for restoring and saving checkpoints.

Instead of using counter, use [`globel_step`](https://www.tensorflow.org/tutorials/mnist/tf/#training) to keep track of steps:

```python
global_step = tf.Variable(0, name='global_step', trainable=False)
train_op = optimizer.minimize(loss, global_step=global_step)
```
Then you can save with:

```python
saver.save(sess, save_path, global_step=global_step)
```

When you restore, the value of global_step is restored as well :-D

**Comment by nightrome** [comment]
But keep in mind that we have two optimizers (for gen. and discr.) and gen. is updated twice, whereas discr. is only updated once per minibatch. So if you want to continue training from a checkpoint, both optimizers would need different global_step values. (the step determines e.g. the learning rate)

**Comment by LarsDu** [comment]
How the heck to you reload global_step?
Everytime I load a model it seems this op's values goes back to 0

**Comment by haoliangjiang** [comment]
> How the heck to you reload global_step?
> Everytime I load a model it seems this op's values goes back to 0

I guess you need to get it from the model_name.
